tests/integration/trainerMissions.test.js

```diff
@@ -1,331 +1,361 @@
-//     const roles = [
-//       { name: 'helper', expectedCode: 403 },
-//       { name: 'planning_referent', expectedCode: 403 },
-//       { name: 'client_admin', expectedCode: 403 },
-//       { name: 'trainer', expectedCode: 403 },
-//     ];
-//     roles.forEach((role) => {
-//       it(`should return ${role.expectedCode} as user is ${role.name}`, async () => {
-//         authToken = await getToken(role.name);
-
-//         const courses = [courseList[0]._id.toHexString(), courseList[1]._id.toHexString()];
-//         const formData = {
-//           trainer: trainer._id.toHexString(),
-//           file: 'test',
-//           fee: 1200,
-//         };
-
-//         const form = generateFormData(formData);
-//         courses.forEach(course => form.append('courses', course));
-
-//         uploadCourseFileStub.returns({ publicId: '1234567890', link: 'https://test.com/file.pdf' });
-
-//         const response = await app.inject({
-//           method: 'POST',
-//           url: '/trainermissions',
-//           headers: { ...form.getHeaders(), Cookie: `alenvi_token=${authToken}` },
-//           payload: getStream(form),
-//         });
-
-//         expect(response.statusCode).toBe(role.expectedCode);
-//       });
-//     });
-//   });
-// });
+describe('TRAINER MISSIONS ROUTES - POST /trainermissions', () => {
+  let authToken;
+  let uploadCourseFileStub;
+
+  beforeEach(async () => {
+    await populateDB();
+    uploadCourseFileStub = sinon.stub(GCloudStorageHelper, 'uploadCourseFile')
+      .returns({ publicId: '123', link: 'ceciestunlien' });
+  });
+  afterEach(() => {
+    uploadCourseFileStub.restore();
+  });
+
+  describe('TRAINING_ORGANISATION_MANAGER', () => {
+    beforeEach(async () => {
+      authToken = await getToken('training_organisation_manager');
+    });
+
+    it('should upload trainer mission for a single course', async () => {
+      const formData = {
+        courses: courseList[0]._id.toHexString(),
+        trainer: trainer._id.toHexString(),
+        file: 'test',
+        fee: 0,
+      };
+      const form = generateFormData(formData);
+
+      uploadCourseFileStub.returns({ publicId: '1234567890', link: 'ceciestunautrelien' });
+
+      const response = await app.inject({
+        method: 'POST',
+        url: '/trainermissions',
+        headers: { ...form.getHeaders(), Cookie: `alenvi_token=${authToken}` },
+        payload: getStream(form),
+      });
+
+      expect(response.statusCode).toBe(200);
+      const trainerMissionCount = await TrainerMission.countDocuments({
+        courses: [courseList[0]._id],
+        date: CompaniDate().startOf(DAY).toISO(),
+        trainer: trainer._id,
+        fee: 0,
+        file: { publicId: '1234567890', link: 'ceciestunautrelien' },
+        creationMethod: UPLOAD,
+      });
+      expect(trainerMissionCount).toBe(1);
+    });
+
+    it('should upload trainer mission when there\'s already a trainer mission for trainer in single course',
```

Comment:
should upload trainer mission when there\'s already a trainer mission for another trainer